src/lib/one-graph/client.js

```diff
@@ -0,0 +1,835 @@
+  const result = await fetchOneGraph(null, appId, internalOperationsDoc, 'CreatePersistedQueryMutation', {
+    nfToken: netlifyToken,
+    appId,
+    query: document,
+    tags,
+    description,
+  })
+
+  const persistedQuery =
+    result.data &&
+    result.data.oneGraph &&
+    result.data.oneGraph.createPersistedQuery &&
+    result.data.oneGraph.createPersistedQuery.persistedQuery
+
+  return persistedQuery
+}
+
+const refetchAndGenerateFromOneGraph = async ({ netlifyToken, netligraphConfig, siteId, state }) => {
+  await ensureAppForSite(netlifyToken, siteId)
+
+  const enabledServicesInfo = await fetchEnabledServices(netlifyToken, siteId)
+  if (!enabledServicesInfo) {
+    console.warn('Unable to fetch enabled services for site for code generation')
+    return
+  }
+
+  const enabledServices = enabledServicesInfo
+    .map((service) => service.service)
+    .sort((aString, bString) => aString.localeCompare(bString))
+  const schema = await fetchOneGraphSchema(siteId, enabledServices)
+  let currentOperationsDoc = readGraphQLOperationsSourceFile(netligraphConfig)
+
+  if (currentOperationsDoc.trim().length === 0) {
+    currentOperationsDoc = defaultExampleOperationsDoc
+  }
+
+  const parsedDoc = parse(currentOperationsDoc)
+  const operations = extractFunctionsFromOperationDoc(parsedDoc)
+
+  generateFunctionsFile(netligraphConfig, schema, currentOperationsDoc, operations)
+  writeGraphQLSchemaFile(netligraphConfig, schema)
+  state.set('oneGraphEnabledServices', enabledServices)
+}
+
+const updateGraphQLOperationsFile = async ({ authToken, docId, netligraphConfig, schema, siteId }) => {
+  const persistedDoc = await fetchPersistedQuery(authToken, siteId, docId)
+  if (!persistedDoc) {
+    console.warn('No persisted doc found for:', docId)
+    return
+  }
+
+  const doc = persistedDoc.query
+
+  writeGraphQLOperationsSourceFile(netligraphConfig, doc)
+  const appOperationsDoc = readGraphQLOperationsSourceFile(netligraphConfig)
+  const parsedDoc = parse(appOperationsDoc, {
+    noLocation: true,
+  })
+  const operations = extractFunctionsFromOperationDoc(parsedDoc)
+  generateFunctionsFile(netligraphConfig, schema, appOperationsDoc, operations)
+}
+
+const friendlyEventName = (event) => {
+  const { __typename, payload } = event
+  switch (__typename) {
+    case 'OneGraphNetlifyCliSessionTestEvent':
+      return friendlyEventName(payload)
+    case 'OneGraphNetlifyCliSessionGenerateHandlerEvent':
+      return 'Generate handler as Netlify function '
+    case 'OneGraphNetlifyCliSessionPersistedLibraryUpdatedEvent':
+      return `Sync Netlify Graph operations library`
+    default: {
+      return `Unrecognized event (${__typename})`
+    }
+  }
+}
+
+const handleCliSessionEvent = async ({ authToken, event, netligraphConfig, schema, siteId }) => {
+  const { __typename, payload } = await event
+  switch (__typename) {
+    case 'OneGraphNetlifyCliSessionTestEvent':
```

Comment:
As a rule, can case statements be wrapped in a block, for readability and to keep the scope contained. This should be a lint rule really.

Reply:
What does it mean to wrap it in a block? And what scope are we concerned with here?